src/engine-scripts/puppet/jsReady.js

```diff
@@ -0,0 +1,36 @@
+module.exports = async (page) => {
+	await page.emulateMediaFeatures([
+		{ name: 'prefers-reduced-motion', value: 'reduce' },
+	]);
+	await page.evaluate( () => {
+		const skin = mw.config.get( 'skin' );
+		let moduleName;
+		switch( skin ) {
+			case 'vector-2022':
+				moduleName = 'skins.vector.js';
+				break;
+			case 'vector':
+				moduleName = 'skins.vector.legacy.js';
+				break;
+			default:
+				return true;
+		}
+		return new Promise((resolve) => {
+			setInterval(() => {
+				if ( mw.loader.getState( moduleName ) === 'ready' ) {
+					resolve(true);
+				}
+			}, 500 );
+		});
+	});
+
+	// wait for animation frame or two.
+        // Wait until the next frame before resolving.
+	await new Promise( resolve => {
+	  requestAnimationFrame( () => {
+	    requestAnimationFrame( () => {
+	      resolve();
+	    } );
+	  } );
+	} );
```

Comment:
This is currently throwing an error because it needs to go in a `page.evaluate`, but I'm also wondering if it's needed at this point? Can it be removed or have you had issues without it?

Reply:
Even with the removal of animations we still use requestAnimationFrame for the resize to get it off the critical path so I think this is needed to be 100% sure. Moving it into the evaluate is obviously needed, that's my bad.

I'm wrapping up for the day so feel free to amend while I am out. If not I'll pick this up Monday.